### PR TITLE
Transition from `IAssemblyLoadContext` to `AssemblyLoadContext`.

### PR DESCRIPTION
**Pull request by NTaylorMullen**
- `IAssemblyLoadContext` is a DNX defined type that will eventually go away.

#3365

### CODE REVIEW

**Comment by pranavkm** [comment]
:shipit:

**Comment by davidfowl** [comment]
:shipit:

File: src/Microsoft.AspNet.Mvc.Razor/Compilation/RoslynCompilationService.cs [hunk 1]
```diff
@@ -146,6 +154,15 @@ public CompilationResult Compile(RelativeFileInfo fileInfo, string compilationCo
             }
         }
 
+        private Assembly LoadStream(MemoryStream ms, MemoryStream assemblySymbols)
+        {
+#if NET451
+            return Assembly.Load(ms.ToArray(), assemblySymbols?.ToArray());
+#else
+            return _razorLoadContext.Load(ms, assemblySymbols);
```
Comment (davidfowl) [inline src/Microsoft.AspNet.Mvc.Razor/Compilation/RoslynCompilationService.cs]: Make sure you can pass null here

Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Mvc.Razor/Compilation/RoslynCompilationService.cs]: It's all good.

**Pull request merged**